src/opentimelineio/clip.cpp

```diff
@@ -19,33 +23,117 @@ Clip::~Clip()
 MediaReference*
 Clip::media_reference() const noexcept
 {
-    return _media_reference;
+    auto active = _media_references.find(_active_media_reference_key);
+    return active == _media_references.end() || !active->second
+               ? nullptr
+               : active->second;
+}
+
+Clip::MediaReferences
+Clip::media_references() const noexcept
+{
+    MediaReferences result;
+    for (auto const& m: _media_references)
+    {
+        result.insert(
+            { m.first, dynamic_retainer_cast<MediaReference>(m.second) });
+    }
+    return result;
+}
+
+template <typename MediaRefMap>
+bool
+Clip::check_for_valid_media_reference_key(
+    std::string const& key,
+    MediaRefMap const& media_references,
+    ErrorStatus*       error_status)
+{
+    auto found = media_references.find(key);
+    if (found == media_references.end())
+    {
+        if (error_status)
+        {
+            *error_status = ErrorStatus(
+                ErrorStatus::MEDIA_REFERENCES_MISSING_ACTIVE_KEY,
+                "The media references do not contain the active key",
+                this);
+        }
+        return false;
+    }
+    return true;
```

Comment:
Should there be a second error status of `Active key exists but has no media references`?

Reply:
Hmm. The find on line ensure its has some kind of value (since a key must have a value), and a nullptr will be converted into a MissingReference later on in the setter.  Is there another type of invalid value we should be checking for that I am not thinking of?

Reply:
MissingReference will be returned